src/main/java/org/mskcc/oncokb/curation/util/parser/ProteinChangeParser.java

```diff
@@ -0,0 +1,360 @@
+        Matcher m = p.matcher(proteinChange);
+
+        ParsingStatus<Alteration> parsingStatus = new ParsingStatus<>();
+        if (m.matches()) {
+            Alteration alteration = new Alteration();
+            String ref = Optional.ofNullable(m.group(1)).orElse("").toUpperCase();
+            alteration.setStart(Integer.valueOf(m.group(2)));
+            if (m.group(5) != null) {
+                alteration.setEnd(Integer.valueOf(m.group(5)));
+            } else {
+                alteration.setRefResidues(ref);
+                alteration.setEnd(alteration.getStart());
+            }
+
+            String revisedProteinChange = ref + alteration.getStart();
+            if (m.group(3) != null) {
+                revisedProteinChange += m.group(3).toUpperCase();
+            }
+            if (m.group(4) != null) {
+                revisedProteinChange += m.group(4).toUpperCase();
+            }
+            revisedProteinChange += "fs";
+            if (m.group(6) != null) {
+                revisedProteinChange += m.group(6);
+            }
+
+            Consequence consequence = new Consequence();
+            consequence.setTerm(FRAMESHIFT_VARIANT.name());
+            alteration.setConsequence(consequence);
+
+            alteration.setProteinChange(StringUtils.isEmpty(revisedProteinChange) ? proteinChange : revisedProteinChange);
+            parsingStatus.setEntity(alteration);
+            parsingStatus.setStatus(EntityStatusType.OK);
+        }
+        return parsingStatus;
+    }
+
+    public static ParsingStatus<Alteration> parseRange(String proteinChange) {
+        Pattern p = Pattern.compile("([A-Z]?)([0-9]+)_([A-Z]?)([0-9]+)(.+)", CASE_INSENSITIVE);
+        Matcher m = p.matcher(proteinChange);
+        ParsingStatus<Alteration> parsingStatus = new ParsingStatus<>();
+        if (m.matches()) {
+            Alteration alteration = new Alteration();
+            String revisedProteinChange = "";
+
+            alteration.setStart(Integer.valueOf(m.group(2)));
+            alteration.setEnd(Integer.valueOf(m.group(4)));
+            String variant = m.group(5);
+
+            HashMap<String, MutationConsequence> termsToCheck = new HashMap<>();
+            termsToCheck.put("mis", MISSENSE_VARIANT);
+            termsToCheck.put("ins", INFRAME_INSERTION);
+            termsToCheck.put("del", INFRAME_DELETION);
+            termsToCheck.put("fs", FEATURE_TRUNCATION);
+            termsToCheck.put("trunc", FEATURE_TRUNCATION);
+            termsToCheck.put("dup", INFRAME_INSERTION);
+            termsToCheck.put("mut", ANY);
+
+            MutationConsequence mutationConsequence = termsToCheck.get(variant);
+            if (mutationConsequence != null) {
+                revisedProteinChange += m.group(1).toUpperCase();
+                revisedProteinChange += m.group(2) + "_";
+                if (m.group(3) != null) {
```

Comment:
Since group 3 is `([A-Z]?)`, I believe it will return empty string if not present instead of null. If it was written as `([A-Z])?` then group 3 can be null.

Can remove this null check.